src/main/kotlin/no/nav/familie/ef/sak/vedtak/PatchPeriodetypeBarnetilsynController.kt

```diff
@@ -0,0 +1,52 @@
+package no.nav.familie.ef.sak.vedtak
+
+import no.nav.familie.ef.sak.vedtak.domain.BarnetilsynWrapper
+import no.nav.familie.ef.sak.vedtak.domain.Periodetype
+import no.nav.familie.ef.sak.vedtak.dto.ResultatType
+import no.nav.security.token.support.core.api.Unprotected
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.http.MediaType
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping(path = ["/api/periodetype-barnetilsyn-patch"], produces = [MediaType.APPLICATION_JSON_VALUE])
+@Unprotected
+@Validated
+class PatchPeriodetypeBarnetilsynController(
+    private val vedtakRepository: VedtakRepository
+) {
+
+    val logger: Logger = LoggerFactory.getLogger(this::class.java)
+
+    @Transactional
+    @PostMapping("{dryRun}")
+    fun patchPeriodetyperBarnetilsyn(@PathVariable(required = false) dryRun: Boolean? = true) {
+
+        val barnetilsynSaker = vedtakRepository.findAll().filter { it.barnetilsyn != null }
+
+        if (barnetilsynSaker.any { it.resultatType == ResultatType.SANKSJONERE }) {
+            error("Skal ikke finnes sanksjon av barnetilsyn")
+        }
+
+        val patchetSaker = barnetilsynSaker.map { vedtak ->
+            val nyePerioder = vedtak.barnetilsyn?.perioder?.map { periode ->
+                periode.copy(periodetype = if (periode.erMidlertidigOpphør == true) Periodetype.OPPHØR else Periodetype.ORDINÆR)
+            } ?: error("Fant ingen perioder")
+
+            vedtak.copy(barnetilsyn = BarnetilsynWrapper(perioder = nyePerioder, begrunnelse = vedtak.barnetilsyn.begrunnelse))
+        }
```

Comment:
Er det hensiktsmessig å logge hvordan vedtakene ser ut før og etter mapping her? Kanskje dette ikke har noe for seg siden du har laget tester på dette, men da kan man verifisere at det ser riktig ut under en dryrun, kanskje?